### ISSUE TITLE
Auto-mute/unmute desktop audio

### ISSUE THREAD

**Issue by Wengier** [SamDel/ChromeCast-Desktop-Audio-Streamer#45]
Right now I have to mute and unmute the computer audio every time after selecting or unselecting a device to stream. Could you please add the option to automatically mute/unmute the audio on the computer when streaming starts/ends? Thanks!

**Comment by SamDel** [comment]
Nice option, thanks! I'll try to implement it later.

**Comment by SamDel** [comment]
Can you try [Setup 2.4.1.zip](https://github.com/SamDel/ChromeCast-Desktop-Audio-Streamer/files/3342278/Setup.2.4.1.zip)?

**Comment by FA-Bubba** [comment]
File Dates are OK in v2.4.1

On install, I was prompted if I wanted to Repair or Uninstall -- I chose Repair.

**Comment by SamDel** [comment]
Thanks.

2.4.1 has a new option: "Auto-mute/unmute desktop audio" on the options tab:
- When you start a stream the desktop audio is muted.
- When you stop all streams the desktop audio is unmuted.

@Wengier requested it and I think I'm going to use it myself.

**Comment by FA-Bubba** [comment]
I'm not sure what this means...

When I am running the Streamer App, and activate speakers, I believe I am "starting a stream".  Since my source is from iTunes running on my PC, I consider that stream is my 'desktop audio'...

If 'desktop audio' gets muted when I start it, then no audio is going to my speakers?

I must be misunderstanding something.

**Comment by Wengier** [comment]
@SamDel: Thanks for adding this option, which makes it even more convenient to use.

@FA-Bubba: It means that the sound will always come from EITHER the computer OR the speaker, but not both at the same time. Hope this helps.

**Comment by SamDel** [comment]
In 2.4.1 the new option doesn't work in combination with the 'Automatically start ... at startup' options. That will be fixed in the next version.

@FA-Bubba : Maybe this isn't clear to you: When the desktop speakers are muted the audio that's playing can still be captured and streamed!

**Comment by FA-Bubba** [comment]
@Wengier; @SamDel...
Thank you, That was my second guess;  Since I have an external speaker plugged into the PC's audio port, and I turn it off when I am running the Streamer App, I may not use that feature, but it's good to know about in case I decide to cast from a different PC...

**Comment by FA-Bubba** [comment]
I tried v2.4.1, and the App did not find all of my speakers.  I will try again later this evening, in case it is something on that PC or my network causing the fail.  Log is attached...


[Log-2-4-1-Fail.zip](https://github.com/SamDel/ChromeCast-Desktop-Audio-Streamer/files/3346552/Log-2-4-1-Fail.zip)

**Comment by SamDel** [comment]
It looks like network problems.

**Comment by FA-Bubba** [comment]
I sort of thought it might have been a network issue, as I have had some of those in the past.

Can you suggest some key 'phrases' I can look for in the Log that would help me identify network issues?
...my goal would be to identify the timing of when it occurred to correlate with any other activities that may be causative factors.

Thanks!

**Comment by kalyway101** [comment]
I think this is a great addition, but I'm curious... is all audio captured based on the volume of the capture device? For example, it works well, albeit a bit laggy even when using the lowest quality, but it does work.

My "issue" is that it only works when my laptop volume is turned up and it can actually capture audio. Is this the intended purpose?

**Comment by SamDel** [comment]
I'm not sure, but I think I had the same behaviour long time ago: changing the device volume changed the volume of the stream. I don't know what changed, maybe I was running Windows 8 back then or the configuration changed.

Now, for me, changing the volume/mute of the device has no effect on the volume of the stream.

Something must be difefrent for you then. What's your setup? Do you use the default laptop speakers? Running Windows 10?

**Comment by FA-Bubba** [comment]
FYI, between my PC, iTunes, and the Streamer App, there are several "volume controls", but some are linked:
 - On my PC, there is the PC Volume control & the "Dell Audio" control, and both are linked, so adjusting one automatically moves the other control in sync, but does not affect the volume of the audio stream, but will change the volume of the PC speakers (or any speakers connected to the PC's Audio Out Port).

 - There is a volume control on the music source (iTunes, WinAmp, or Media Player, ..., etc), and adjusting this control does change the volume of the streamed music.

 - In Google Home, there are volume controls for each speaker, and changing that also changes the volume bar in each respective Device Box in the Streamer App, and vice-versa...

So basically, there seems to be just two ways to adjust the streamed audio volume:
1. In the Source App (iTunes, CDJ, Media Player, WinAmp, ..., etc) -- adjusts all speakers together.
2. In the Device Boxes in the Streamer APP --OR-- in Google Home -- adjusts individual speakers (unless you are adjusting a Group).

Of course, if you have volume controls on each speaker, they work independently.

**Comment by SamDel** [comment]
Thanks, the auto-mute/unmute works fine here!
